### DISCUSSION TITLE
Is here a direct way to read text file which is not encoded by utf-8

### DISCUSSION THREAD

**Discussion by donald001** [vlang/v#20979] [Questions and Answers]
Is here a direct way to read text file which is not encoded by utf-8? 
You know in java we can read it like `List<String> lines = Files.readAllLines(path, StandardCharsets.UTF_16);`. 
I can only `os.read_lines(path)` in V. But don't know where to specify the encoding charsets.

**Comment by danabooth** [comment]
I don't mess around with V too often, but would something like this work, but sub the ```[]u8``` with whatever you want:
```
mut buf := []u8{len: 4096}
mut infile := os.open('<filename>')!
for {
    n := infile.read(mut &buf) or {break}
    <do something>
}
infile.close()
```

**Reply by JalonSolov** [reply]
V doesn't differentiate.  It reads files as a series of bytes.  If those bytes are a UTF-8 string, or a plain ASCII string, fine, it's still just a series of bytes.

However, you are correct that it doesn't support specifying charsets.

**Reply by donald001** [reply]
Yes, but for text files, without specifying the charsets, we got the wrong text. For example,  the raw file with "UTF-16" encoded is 
![image](https://github.com/vlang/v/assets/38281146/2cd77ba6-ced1-4fe0-ae8b-03d6ae4fecf6)
However, using `os.read_file` or `os.read_lines`, we just get the wrong text: 
![image](https://github.com/vlang/v/assets/38281146/421d66d7-63fa-40cf-be2a-1d1c25c40796)

**Reply by JalonSolov** [reply]
True.  And while V does have support for both UTF-8 and UTF-32, it doesn't have support for UTF-16.

**Reply by donald001** [reply]
Okay and thanks, we hope it could someday. 
Utf-16 contains "UTF_16BE" and "UTF_16LE". Also it would be nice if "ISO_8859_1" will be taken into consideration. They are all `StandardCharsets` in Java^_^.
